sdk/core/src/parsing.rs

```diff
@@ -42,6 +42,55 @@ pub fn from_azure_time(s: &str) -> Result<chrono::DateTime<chrono::Utc>, chrono:
     Ok(dt_utc)
 }
 
+pub mod rfc2822_time_format {
+    use super::from_azure_time;
+    use chrono::{DateTime, Utc};
+    use serde::{self, Deserialize, Deserializer, Serializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<DateTime<Utc>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        from_azure_time(&s).map_err(serde::de::Error::custom)
+    }
+
+    pub fn serialize<S>(date: &DateTime<Utc>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let s = format!("{}", date.to_rfc2822());
```

Comment:
Learning q: what's the point of doing this since it's just basically a to-string?

Reply:
Unfortunately `chrono` datetimes are serialized as rfc3339 while Azure uses rfc2822. With this module we can decorate the datetime fields with `#[serde(with = "azure_core::parsing::rfc2822_time_format")]` and get the proper Azure representation.

Reply:
Oh, I get that. I meant specifically, why `format` the result of `to_fc2822`? Wouldn't that returned an owned string anyway, or does it not and, hence, formatting is an easy way to create an owned string. At a glance, it seems like a waste, like in C#:

```c#
string foo = "foo";
string foo2 = string.Format("{0}", foo);
```
There's almost no reason (certainly no practical reason) to do this. Formatting in .NET is also expensive. Perhaps not in Rust, hence why I "tagged" this as a learning question.

Reply:
Ah, sorry I misunderstood your question 🙏 !

Yes, you are absolutely right, there formatting a string like that is a waste! I'll fix it ASAP!

Reply:
Done! Thank you for noticing this! 👍